pkg/controllers/edgeconnect/controller_test.go

```diff
@@ -718,6 +738,18 @@ func mockNewEdgeConnectClientCreate(edgeConnectClient *edgeconnectmock.Client, h
 			nil,
 		)
 
+		v := "my:default"
+		edgeConnectClient.On("GetConnectionSetting", mock.Anything).Return(edgeconnect.EnvironmentSetting{
+			ObjectId:      &v,
+			SchemaId:      edgeconnect.KubernetesConnectionSchemaID,
+			SchemaVersion: edgeconnect.KubernetesConnectionVersion,
+			Scope:         edgeconnect.KubernetesConnectionScope,
+			Value:         edgeconnect.EnvironmentSettingValue{},
+		}, nil).Maybe()
+		edgeConnectClient.On("CreateConnectionSetting", mock.Anything).Return(nil).Maybe()
+		edgeConnectClient.On("UpdateConnectionSetting", mock.Anything).Return(nil).Maybe()
+		edgeConnectClient.On("DeleteConnectionSetting", mock.Anything).Return(nil).Maybe()
```

Comment:
can we avoid the use of `Maybe()`, and mock the client for the exact scenario we are testing?

Reply:
☝️  very good point

Reply:
Done [here](https://github.com/Dynatrace/dynatrace-operator/pull/3206/commits/9ab3b9530311f58228cd4bfe687929fba222a126)